packages/components/src/DataList/DataList.tsx

```diff
@@ -55,5 +64,32 @@ export function DataList<T extends DataListObject>({
   );
 }
 
+function ResultsCounter({
```

Comment:
I'd put this in its own file. I see the appeal of putting everything in the same file but it's also a slippery slope into a gigantic file that is harder to maintain. Remember the phrase "One file, one component".

Also, there's something about the name that feels misleading. It isn't really counting anything internally. It's more presentation. What about `TotalCount` or `DataListTotalCount` if you want to follow the pattern under the components folder?

Reply:
Agreed! Moved it. Thanks for the suggestion.

I created it on the main file because it was not actually being exported as `DataListLayout` or `EmptyState`, but it does make sense to keep one component per file to keep it organized.